source/code/plugins/scom_configuration.rb

```diff
@@ -0,0 +1,100 @@
+module SCOM
+
+  class Configuration
+    require 'openssl'
+    require 'uri'
+        
+    require_relative 'oms_configuration'
+    require_relative 'omslog'
+        
+    @@configuration_loaded = false
+    @@scom_conf_path = '/etc/opt/microsoft/omsagent/scom/conf/omsadmin.conf'
+    @@cert_path = '/etc/opt/microsoft/omsagent/scom/certs/scom-cert.pem'
+    @@key_path =  '/etc/opt/microsoft/omsagent/scom/certs/scom-key.pem'     
+  
+    @@cert = nil
+    @@key = nil
+        
+    @@scom_endpoint = nil
+    @@scom_event_endpoint = nil
+    @@scom_perf_endpoint = nil
+    
+    @@monitoring_id = nil    
+      
+    def self.load_scom_configuration
+      return true if @@configuration_loaded
+      return false if !OMS::Configuration.test_onboard_file(@@scom_conf_path) or !OMS::Configuration.test_onboard_file(@@cert_path) or !OMS::Configuration.test_onboard_file(@@key_path)
+            
+      scom_endpoint_lines = IO.readlines(@@scom_conf_path).select{ |line| line.start_with?("SCOM_ENDPOINT")}
+      if scom_endpoint_lines.size == 0
+        OMS::Log.error_once("Could not find SCOM_ENDPOINT in #{conf_path}")
+        return false
+      elsif scom_endpoint_lines.size == 0
+        OMS::Log.warn_once("Found more than one SCOM_ENDPOINT setting in #{conf_path}, will use the first one.")
+      end 
+            
+      begin
+        scom_endpoint_url = scom_endpoint_lines[0].split("=")[1].strip
+        @@scom_endpoint = URI.parse( scom_endpoint_url )
+        @@scom_event_endpoint = @@scom_endpoint.clone
+        @@scom_event_endpoint.path = '/RESTWCFServiceLibrary/events'
+        @@scom_perf_endpoint = @@scom_endpoint.clone
+        @@scom_perf_endpoint.path = '/RESTWCFServiceLibrary/perf'
+      rescue => e
+        OMS::Log.error_once("Error parsing endpoint url. #{e}")
+        return false
+      end
+  
+      monitoring_id_lines = IO.readlines(@@scom_conf_path).select{ |line| line.start_with?("MONITORING_ID")}
+      if monitoring_id_lines.size == 0
+        OMS::Log.error_once("Could not find MONITORING_ID in #{@@scom_conf_path}")
+        return false
+      elsif monitoring_id_lines.size == 0
+        OMS::Log.warn_once("Found more than one MONITORING_ID setting in #{@@scom_conf_path}, will use the first one.")
+      end
```

Comment:
This block is the same (except for the value being searched for) as lines 28-34. Please extract into a helper method.